src/client/dfuse/il/int_posix.c

```diff
@@ -34,6 +34,7 @@
 #include "dfuse_common.h"
 
 #include "ioil.h"
+#include "../pil4dfs/hook.h"
```

Comment:
this doesn't look nice to me.
im not sure the old ioil should depend on this hook for pil4dfs. but i need to understand more why this is needed.

Reply:
ok. I can rearrange the location of hook.h and hook.c.
The details can be found in ticket https://daosio.atlassian.net/browse/DAOS-16131.
With current method of function interception, we have dead lock when both libioil and libjemalloc are present.

dlsym calls malloc().
malloc calls dfuse_mmap().
dfuse_mmap() needs dlsym() to query the address of mmap() in libc.so.
Infinite loop.

Reply:
With trampoline method, we can control when the interception is enabled.
The interception is enabled since application process starts with current interception method.

Reply:
i didn't mean we should manage the location of the hook, it's just weird that IOIL needs to depend on this hook which was only for pil4dfs. i guess let's see if @jolivier23 has some thoughts on this.

Reply:
I see.
Jeff previously approved this PR a couple of weeks ago. There was minor changes to fix NLT fault injection issues.
@jolivier23 Could you please review again? Thank you very much!

Reply:
With suggestion above, you can just do #include <pil4dfs/hook.h>

Reply:
Alternatively, one could move "hook.h" to a common area

Reply:
is it still planned to relocate the hook.h and hook.c ?
From my side, I would respectively move it in `src/daos` and `src/common` directories as such facilities could be useful in other part of the DAOS code.
@jolivier23 and @wiliamhuang , does it make sense for you.

Reply:
Since hook.c and hook.h are only for trampoline interception, they probably will be used by only libioil and libpil4dfs. 
Shall we create a directory src/client/dfuse/trampoline and put them under "trampoline"?

Reply:
From my side, it could have been helpful to have such facilities for solving ticket [DAOS-14532](https://daosio.atlassian.net/browse/DAOS-14532): overloading `getenv()` and `setenv()` function with trampoline could have been a solution.
This is why, I was thinking it could make sense to not limit the hook facilities to the client code.
However, it is just a suggestion coming from my "padawan" experience of DAOS: developer with much more experience with DAOS such as @jolivier23 and @mchaarawi should be have a more relevant opinion on this point.

Reply:
I understand now. Thank you for your example use case.